integration-tests/src/tests/transaction_payment.rs

```diff
@@ -1,4 +1,181 @@
+	);
+
+	PolimecNet::execute_with(|| {
+		let alice: PolimecAccountId = ALICE.into();
+
+		let (block_author, _) = &constants::collators::invulnerables()[1];
+		// Block author's aura index is 1
+		set_author(1u64);
+		assert_eq!(polimec_runtime::Authorship::author(), Some(block_author.clone()));
+
+		let usdt_id = AcceptedFundingAsset::USDT.id();
+		let usdt_decimals = <PolimecForeignAssets as fungibles::metadata::Inspect<PolimecAccountId>>::decimals(usdt_id);
+		let usdt_unit = 10u128.pow(usdt_decimals as u32);
+		let plmc_decimals = PLMC_DECIMALS;
+		let plmc_unit = 10u128.pow(plmc_decimals as u32);
+
+		PolimecBalances::set_balance(&alice, 0u128);
+		PolimecForeignAssets::set_balance(usdt_id, &alice, 100 * usdt_unit);
+		// Fees are usually very small, so we need to give the treasury an ED.
+		PolimecForeignAssets::set_balance(
+			usdt_id,
+			&polimec_runtime::BlockchainOperationTreasury::get(),
+			100 * usdt_unit,
+		);
+		// Block author doesn't need to have any balance, as the tip is bigger than ED.
+
+		let paid_call = PolimecCall::System(frame_system::Call::remark { remark: vec![69, 69] });
+
+		type TxPaymentExtension = pallet_asset_tx_payment::ChargeAssetTxPayment<PolimecRuntime>;
+		let signed_extension =
+			pallet_asset_tx_payment::ChargeAssetTxPayment::<PolimecRuntime>::from(10 * plmc_unit, Some(usdt_id));
+
+		let dispatch_info = paid_call.get_dispatch_info();
+		let FeeDetails { inclusion_fee, tip } =
+			polimec_runtime::TransactionPayment::compute_fee_details(0u32, &dispatch_info, 10u128 * plmc_unit);
+		let expected_plmc_fee = inclusion_fee.expect("call should charge a fee").inclusion_fee();
+		let expected_plmc_tip = tip;
+
+		let plmc_price_decimal_aware =
+			<PriceProviderOf<PolimecRuntime>>::get_decimals_aware_price(PLMC_FOREIGN_ID, USD_DECIMALS, plmc_decimals)
+				.expect("Price irretrievable");
+
+		// USDT should be configured with the same decimals as our underlying USD unit, and we set the price to 1USD at the beginning of this test.
+		let expected_usd_fee = plmc_price_decimal_aware.saturating_mul_int(expected_plmc_fee);
+		let expected_usd_tip = plmc_price_decimal_aware.saturating_mul_int(expected_plmc_tip);
+
+		let prev_alice_usdt_balance = PolimecForeignAssets::balance(usdt_id, alice.clone());
+		let prev_alice_plmc_balance = PolimecBalances::balance(&alice);
+		let prev_blockchain_operation_treasury_usdt_balance =
+			PolimecForeignAssets::balance(usdt_id, polimec_runtime::BlockchainOperationTreasury::get());
+		let prev_blockchain_operation_treasury_plmc_balance =
+			PolimecBalances::balance(&polimec_runtime::BlockchainOperationTreasury::get());
+		let prev_block_author_usdt_balance = PolimecForeignAssets::balance(usdt_id, block_author.clone());
+		let prev_block_author_plmc_balance = PolimecBalances::balance(&block_author.clone());
+
+		let pre = signed_extension.pre_dispatch(&alice, &paid_call, &dispatch_info, 8).unwrap();
```

Comment:
**Q:** Why is the `len` 8 here?

Reply:
Not sure, I think it was failing with 0. Ill investigate lengths further